ndb/tests/unit/test_model.py

```diff
@@ -1317,11 +1318,182 @@ def test_constructor():
+        expected = "_CompressedValue(" + repr(z_val) + ")"
+        assert repr(compressed_value) == expected
+
+    @staticmethod
+    def test___eq__():
+        z_val1 = zlib.compress(b"12345678901234567890")
+        compressed_value1 = model._CompressedValue(z_val1)
+        z_val2 = zlib.compress(b"12345678901234567890abcde\x00")
+        compressed_value2 = model._CompressedValue(z_val2)
+        compressed_value3 = unittest.mock.sentinel.compressed_value
+        assert compressed_value1 == compressed_value1
+        assert not compressed_value1 == compressed_value2
+        assert not compressed_value1 == compressed_value3
+
+    @staticmethod
+    def test___ne__():
+        z_val1 = zlib.compress(b"12345678901234567890")
+        compressed_value1 = model._CompressedValue(z_val1)
+        z_val2 = zlib.compress(b"12345678901234567890abcde\x00")
+        compressed_value2 = model._CompressedValue(z_val2)
+        compressed_value3 = unittest.mock.sentinel.compressed_value
+        assert not compressed_value1 != compressed_value1
+        assert compressed_value1 != compressed_value2
+        assert compressed_value1 != compressed_value3
+
+    @staticmethod
+    def test___hash__():
+        z_val = zlib.compress(b"12345678901234567890")
+        compressed_value = model._CompressedValue(z_val)
+        with pytest.raises(TypeError):
+            hash(compressed_value)
+
+
+class TestBlobProperty:
+    @staticmethod
+    def test_constructor_defaults():
+        prop = model.BlobProperty()
+        # Check that only one of the constructor defaults was used.
+        assert prop.__dict__ == {"_compressed": False}
+
+    @staticmethod
+    def test_constructor_explicit():
+        prop = model.BlobProperty(
+            name="blob_val",
+            compressed=True,
+            indexed=False,
+            repeated=False,
+            required=True,
+            default=b"eleven\x11",
+            choices=(b"a", b"b", b"c", b"eleven\x11"),
+            validator=TestProperty._example_validator,
+            verbose_name="VALUE FOR READING",
+            write_empty_list=False,
+        )
+        assert prop._name == b"blob_val" and prop._name != "blob_val"
+        assert not prop._indexed
+        assert not prop._repeated
+        assert prop._required
+        assert prop._default == b"eleven\x11"
+        assert prop._choices == frozenset((b"a", b"b", b"c", b"eleven\x11"))
+        assert prop._validator is TestProperty._example_validator
+        assert prop._verbose_name == "VALUE FOR READING"
+        assert not prop._write_empty_list
+
+    @staticmethod
+    def test_constructor_compressed_and_indexed():
+        with pytest.raises(NotImplementedError):
+            model.BlobProperty(name="foo", compressed=True, indexed=True)
+
+    @staticmethod
+    def test__value_to_repr():
+        prop = model.BlobProperty(name="blob")
+        as_repr = prop._value_to_repr(b"abc")
+        assert as_repr == "b'abc'"
```

Comment:
ISTM this is not what was intended for the original

Reply:
What was intended?

Reply:
Well, what I mean to say is that the `repr()` has changed across Python versions:

```
$ python2.7 -c 'print(repr(b"abc"))'
'abc'
$ python3.7 -c 'print(repr(b"abc"))'
b'abc'
```

Reply:
Ok, I wouldn't worry about that.

Reply:
:+1: